helpers.sh

```diff
@@ -174,11 +174,7 @@ btcpay_restart() {
 btcpay_dump_db() {
     pushd . > /dev/null
     cd "$(dirname "$BTCPAY_ENV_FILE")"
-    backup_dir="/var/lib/docker/volumes/backup_datadir/_data"
-    if [ ! -d "$backup_dir" ]; then
-        docker volume create backup_datadir
-    fi
-    local filename=${1:-"postgres-$(date "+%Y%m%d-%H%M%S").sql"}
-    docker exec $(docker ps -a -q -f "name=postgres_1") pg_dumpall -c -U postgres > "$backup_dir/$filename"
+    local file_path=${1:-"postgres-$(date "+%Y%m%d-%H%M%S").sql"}
+    docker exec $(docker ps -a -q -f "name=postgres_1") pg_dumpall -c -U postgres > "$file_path"
```

Comment:
Can we dump directly to a `.sql.gz` file? If the DB is huge, you could run out of disk space during the backup process. Usually it's as easy as adding a pipe to `gzip --best` or `gzip --fast` and piping again when importing.
Regular sql files can be huge so I usually avoid them at all costs and work directly with a compressed file in 1 go.
Might be nitpicking though as BTCPay DBs are not that big most of the time.

Reply:
Agree would defenitely do it, does not cost much but is a saver if the db gets big as the compression is very good even in standard mode. 20GB mysql db dump is ~ 1GB gzipped.

gzip on dump:
`dumpcommand | gzip > file_path`

So for above should be something like this (untested, yolo):
```
local file_path=${1:-"postgres-$(date "+%Y%m%d-%H%M%S").sql.gz"}
docker exec $(docker ps -a -q -f "name=postgres_1") pg_dumpall -c -U postgres | gzip > "$file_path"
```

on restore gunzip and import:
`gunzip < path_to.sql.gz | db_import command`

Reply:
One thing to be mindful of here is the exit code. If the dump command fails, you will be creating a empty `.sql.gz` file and the backup script won't know something failed, so this can be a serious issue.
The file would contain the error message and be very small.

I can think of 2 solutions:
- Checking the file size. Should be more than a few kB (ugly check)
- Maybe we can pipe only the stdout and leave stderr alone? Not sure if possible and what it would look like. Also depends on the dump command if it respects stdout and stderr.

Reply:
Done in [05d96f4](https://github.com/btcpayserver/btcpayserver-docker/pull/641/commits/05d96f4887a91781cb61c894b7b4e8bef4a17a8b).

Reply:
Did you check what happens if the dump fails? Does the backup crash or do you get a dump file containing an error (and not a working backup)?

Reply:
I've now added [fa503a5](https://github.com/btcpayserver/btcpayserver-docker/pull/641/commits/fa503a50c4af1b50d571748c8f9e2a0ab3863910), which ensures that. To my knowledge this is the only way to achieve this, at least if we want to circumvent saving the unzipped version separately. ([details](https://unix.stackexchange.com/questions/385089/how-to-gzip-only-if-mysqldump-does-not-error/451044))